.github/workflows/build-docs.yml

```diff
@@ -0,0 +1,39 @@
+name: Build and Deploy Docs
+
+on:
+  release:
+    types: [published]
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Initialize the environment
+        uses: ./.github/actions/initialize
```

Comment:
Would it make sense to have a different build for the documentation build?  There's no real purpose to having Sphinx et al. installed during testing if the doc build it is a separate step.

Reply:
You mean breaking out some file like `doc_requirements.txt`? That's not a terrible idea, although I don't see the harm of it being installed.

Reply:
Minor optimization.  Install 5x, use once is slower than install once, use once.  Non-blocking / future-work.

Reply:
This is only run once, using whatever their default version of Python is.